File: backend/user-service/src/main/java/com/finance/userservice/service/AccountService.java
package com.finance.userservice.service;

import com.finance.userservice.dto.request.AccountRequest;
import com.finance.userservice.dto.response.AccountResponse;

import java.util.List;

public interface AccountService {
    AccountResponse createAccount(AccountRequest request);
    AccountResponse updateAccount(Long id, AccountRequest request);
    void deleteAccount(Long id);
    AccountResponse getAccount(Long id);
    List<AccountResponse> getAllAccounts();
    List<AccountResponse> getActiveAccounts();
    void deactivateAccount(Long id);
    void activateAccount(Long id);
} 

File: frontend/src/services/accountService.ts
import { api } from './api';
import type { AccountResponse } from '../types/account';
import type { AccountRequest } from '../types/account';

export const accountService = {
    createAccount: async (request: AccountRequest): Promise<AccountResponse> => {
        const response = await api.post<AccountResponse>('/accounts', request);
        return response.data;
    },

    updateAccount: async (id: number, request: AccountRequest): Promise<AccountResponse> => {
        const response = await api.put<AccountResponse>(`/accounts/${id}`, request);
        return response.data;
    },

    deleteAccount: async (id: number): Promise<void> => {
        await api.delete(`/accounts/${id}`);
    },

    getAccount: async (id: number): Promise<AccountResponse> => {
        const response = await api.get<AccountResponse>(`/accounts/${id}`);
        return response.data;
    },

    getAllAccounts: async (): Promise<AccountResponse[]> => {
        const response = await api.get<AccountResponse[]>('/accounts');
        return response.data;
    },

    getActiveAccounts: async (): Promise<AccountResponse[]> => {
        const response = await api.get<AccountResponse[]>('/accounts/active');
        return response.data;
    },

    deactivateAccount: async (id: number): Promise<void> => {
        await api.post(`/accounts/${id}/deactivate`);
    },

    activateAccount: async (id: number): Promise<void> => {
        await api.post(`/accounts/${id}/activate`);
    }
}; 